source/Halibut/Transport/SecureListener.cs

```diff
@@ -86,7 +86,7 @@ public int Start()
 
             log = logFactory.ForEndpoint(new Uri("listen://" + listener.LocalEndpoint));
             log.Write(EventType.ListenerStarted, "Listener started");
-            Task.Run(async () => await Accept());
+            Task.Run(Accept);
```

Comment:
Why was this changed? This will not return until the first await is hit from my understanding

Reply:
This is creating an unnecessary async state machine and object. Task run starts a new thread to handle the call anyway, and then returns immediately.

Reply:
I think you're thinking of the yield state machine.

Reply:
Ah yes, you are correct

Reply:
I think we should change this back to `new Thread(Accept).Run()` now that .NET Core supports the Sync method.